pkg/controller/operators/catalog/operator.go

```diff
@@ -1387,28 +1387,36 @@ func (o *Operator) syncInstallPlans(obj interface{}) (syncError error) {
 
 	querier := o.serviceAccountQuerier.NamespaceQuerier(plan.GetNamespace())
 	ref, err := querier()
-	if err != nil {
-
-		// Retry sync if non-fatal error
-		if !scoped.IsOperatorGroupError(err) {
-			syncError = fmt.Errorf("attenuated service account query failed - %v", err)
-			return
-		}
-
-		// Mark the InstallPlan as failed for a fatal Operator Group related error
-		logger.Infof("attenuated service account query failed - %v", err)
-		ipFailError := fmt.Errorf("invalid operator group - %v", err)
 
-		if err := o.transitionInstallPlanToFailed(plan, logger, v1alpha1.InstallPlanReasonInstallCheckFailed, ipFailError.Error()); err != nil {
-			// retry for failure to update status
+	out := plan.DeepCopy()
+	if err != nil {
+		// Set status condition/message and retry sync if any error
+		ipFailError := fmt.Errorf("attenuated service account query failed - %v", err)
+		logger.Infof(ipFailError.Error())
+		err := o.setInstallPlanInstalledCond(out, v1alpha1.InstallPlanReasonInstallCheckFailed, err.Error(), logger)
+		if err != nil {
 			syncError = err
 			return
 		}
-
-		// Requeue subscription to propagate SubscriptionInstallPlanFailed condtion to subscription
-		o.requeueSubscriptionForInstallPlan(plan, logger)
-
+		syncError = ipFailError
 		return
+	} else {
+		// reset condition/message if it had been set in previous sync. This condition is being reset since any delay in the next steps
+		// (bundle unpacking/plan step errors being retried for a duration) could lead to this condition sticking around, even after
+		// the serviceAccountQuerier returns no error since the error has been resolved (by creating the required resources), which would
+		// be confusing to the user
+
+		// NOTE: this makes the assumption that the InstallPlanInstalledCheckFailed reason is only set in the previous if clause, which is
+		// true in the current iteration of the catalog operator. Any future implementation change that aims at setting the reason as
+		// InstallPlanInstalledCheckFailed must make sure that either this assumption is not breached, or the condition being set elsewhere
+		// is not being unset here unintentionally.
+		if cond := out.Status.GetCondition(v1alpha1.InstallPlanInstalled); cond.Reason == v1alpha1.InstallPlanReasonInstallCheckFailed {
+			err := o.setInstallPlanInstalledCond(out, v1alpha1.InstallPlanConditionReason(corev1.ConditionUnknown), "", logger)
+			if err != nil {
+				syncError = err
+				return
+			}
```

Comment:
After updating the InstallPlan for resetting the condition we're proceeding in the reconcile loop with an outdated (resourceVersion) IP.
And the next step right after this could be to update the status again for the service account reference

https://github.com/operator-framework/operator-lifecycle-manager/blob/2e16b7e83f463f364627f7245adfe07ae631393f/pkg/controller/operators/catalog/operator.go#L1423-L1430

That would result in a resource conflict, sync error, and retry.

So to avoid that, we could either requeue and return here, or update the IP from the return value before proceeding:
```Go
plan, err := o.setInstallPlanInstalledCond(out, v1alpha1.InstallPlanConditionReason(corev1.ConditionUnknown), "", logger)
```

Reply:
@hasbro17 that's a really good point. Although, I was trying to read through what we have in the main branch currently, and doesn't look like we handle this concern anywhere, i.e whenever there's an update, we don't update `plan` to have the latest object. Which means we're always running into the scenario that you've mentioned. And if that's the case, will just updating the plan at this step make any dent at all?

Reply:
I'd have to check the rest of the loop but at least for this part previously we were returning after we had to update the IP to avoid the conflict.
https://github.com/operator-framework/operator-lifecycle-manager/blob/20ded32d2260a8f1eeb594b9ec2147ad0134cfc6/pkg/controller/operators/catalog/operator.go#L1402-L1411

Given that it's an additional request to resolve the conflict I would still lean towards updating or returning.

But it's a trivial issue so we probably don't need to split hairs on it and can punt on this for now 🙂

Reply:
I see, no it makes sense to me. I've updated `plan` with the returned IP after status update to avoid that additional request 👍🏽